File: tests/scripts/import.js

// Check program version
print("Program version:", ovito.version)
assert(ovito.version[0] == "v")

// Check current directory
print("Current directory:", pwd())

// Load test data
obj = load("../files/animation.dump.gz")

// Selection access
assert(ovito.selectedNode != null)
node = ovito.selectedNode

// Clear selection
ovito.selectedNode = null

// Change selection
ovito.selectedNode = node

// Load second file
cd("../files")
obj2 = load("nw2.imd.gz")
assert(obj2.source.sourceUrl == "file:nw*.imd.gz")

// Load third file.
obj2.source.sourceUrl = "SiVacancy.cfg"

// Delete nodes.
obj.deleteNode()
obj2.deleteNode()
assert(ovito.selectedNode == null)

File: tests/scripts/import.py
import ovito
import ovito.io
import ovito.modifiers

# Check program version
print "This is OVITO version", ovito.version

# Load test data
node = ovito.io.import_file("../files/animation.dump.gz")

# Apply a modifier
mod = ovito.modifiers.CoordinationNumberModifier(cutoff = 2.5)
assert mod.cutoff == 2.5
node.modifiers.append(mod)

# Selection access
assert ovito.dataset.selected_node != None
assert ovito.dataset.selected_node == node

# Clear selection
ovito.dataset.selected_node = None

# Load second file
node2 = ovito.io.import_file("../files/nw2.imd.gz")
print node2.source.source_path
assert node2.source.source_path == "../files/nw*.imd.gz"

# Load a third file.
node2.source.source_path = "../files/SiVacancy.cfg"

# Test alternative way of loading a different file:
node2.source.load("../files/shear.void.dump.bin", columns = ["Particle Identifier", "Particle Type", "Position.X", "Position.Y", "Position.Z"])
print node2.source.source_path

# Delete nodes.
node.deleteNode()
node2.deleteNode()
assert ovito.dataset.selectedNode == None
